lms/envs/production.py

```diff
@@ -218,9 +219,22 @@ def get_env_setting(setting):
 AWS_DEFAULT_ACL = 'public-read'
 AWS_BUCKET_ACL = AWS_DEFAULT_ACL
 
-# Change to S3Boto3 if we haven't specified another default storage AND we have specified AWS creds.
-if (not _YAML_TOKENS.get('DEFAULT_FILE_STORAGE')) and AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY:
-    DEFAULT_FILE_STORAGE = 'storages.backends.s3boto3.S3Boto3Storage'
+_yaml_storages = _YAML_TOKENS.get('STORAGES', {})
+
+# For backward compatibility, if YAML provides legacy keys (DEFAULT_FILE_STORAGE, STATICFILES_STORAGE)
+# and STORAGES doesn’t explicitly define the corresponding backend, migrate the legacy value into STORAGES.
+# If YAML doesn't provide lagacy keys, no backend is defined in STORAGES['default'] and AWS creds are present,
+# fall back to S3Boto3Storage.
+#
+# This ensures YAML-provided values take precedence over defaults from common.py,
+# without overwriting user-defined STORAGES and AWS creds are treated only as a fallback.
+if 'DEFAULT_FILE_STORAGE' in _YAML_TOKENS and not _yaml_storages.get('default', {}).get('BACKEND'):
+    STORAGES['default']['BACKEND'] = _YAML_TOKENS['DEFAULT_FILE_STORAGE']
+elif not _yaml_storages.get('default', {}).get('BACKEND') and AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY:
+    STORAGES['default']['BACKEND'] = 'storages.backends.s3boto3.S3Boto3Storage'
```

Comment:
almost good, but consider that the `elif` will only be reached if `not _yaml_storages.get('default', {}).get('BACKEND')` evaluates to True.

Reply:
Yes, the `elif` block will only be executed if `not _yaml_storages.get('default', {}).get('BACKEND')` evaluates to **True** given AWS creds are provided. Is that what you were referring to?

Reply:
Ah, sorry, I was trying to indicate that the `elif` block was redundant, but my logic was wrong. The logic here looks good.

Optional nit -- the whole thing could be wrapped in `if not _yaml_storages.get('default', {}).get('BACKEND')` rather than specifying that condition twice in both the `if` and `elif. That might be easier to read. But it's your call.

Reply:
I’d propose this version just for more clarity
```suggestion
backend_is_missing = not _yaml_storages.get('default', {}).get('BACKEND')
if backend_is_missing:
    if 'DEFAULT_FILE_STORAGE' in _YAML_TOKENS:
        STORAGES['default']['BACKEND'] = _YAML_TOKENS['DEFAULT_FILE_STORAGE']
    elif AWS_ACCESS_KEY_ID and AWS_SECRET_ACCESS_KEY:
        STORAGES['default']['BACKEND'] = 'storages.backends.s3boto3.S3Boto3Storage'

```

Reply:
oh, wait, `elif` will be executed only if `DEFAULT_FILE_STORAGE` is not in `_YAML_TOKENS` or `not _yaml_storages.get('default', {}).get('BACKEND')` evaluates to False.

Reply:
I'd make it private and call it `_storages_default_backend_is_missing` instead, but I agree that's clearer.

Reply:
Makes sense! its updated!